### PR TITLE
Added toolranks support to tools.

### PR DESCRIPTION
**Pull request by Andarius68**

### CODE REVIEW

**Review by Niklp09 — COMMENTED** [review]
- Please add `toolranks` to the mod dependencies (mod.conf)
- I don't think the "toolranks support" comment is needed 6 times, the call to the toolranks API is self-explaining IMO

**Comment by Niklp09** [comment]
@Andarius68 You reverted all changes besides a few newlines, doesn't look right to me :smile:

**Comment by Andarius68** [comment]
Oh, sorry, I actually moved the toolrank support lines back to my own custom mod just for my server. I didn't know there was interest in the changes I made on the fork I made, what with all the comments and what not. 😂 I can put them back though if you wish to use it. Sans the overabundant comments of course. 😉

**Comment by Andarius68** [comment]
I see now that Diablosxm submitted a PR last year that already also adds Toolranks support as well as Technic support so this is all probably a moot point anyway. lol

File: tools.lua [hunk 1]
```diff
@@ -48,6 +48,8 @@ minetest.register_tool("nether:pick_nether", {
 		return itemstack
 	end
 })
+-- Add toolrank support to tools
+toolranks.add_tool("nether:pick_nether")
```
Comment (SmallJoker) [inline tools.lua:52]: `toolranks` is not guaranteed to be loaded at all. Two approached:

**Soft dependency:**
1. Check against `minetest.get_modpath("toolranks")` in a new `minetest.register_on_mods_loaded` callback (before the registrations become read-only)
2. If present: call `toolranks.add_tool`

**Optional dependency:**
1. Check against `minetest.get_modpath("toolranks")` here (at script load time)
2. If present: call `toolranks.add_tool`
3. Add `toolranks` to `mod.conf` ("optional_depends") to ensure that it's loaded before this mod

**Hard dependency:** (not recommended)
1. Add `toolranks` to `mod.conf` ("depends") to ensure that it's loaded before this mod

When touching the dependency list - also update `depends.txt` or remove that file. It's for Minetest <= 0.4.16 compatibility.

**Comment by SmallJoker** [comment]
This PR is very likely superseded by #73 which also introduced technic support.

**Pull request closed without merging**